src/distilabel/pipeline/base.py

```diff
@@ -231,6 +232,8 @@ def __enter__(self) -> Self:
     def __exit__(self, exc_type, exc_value, traceback) -> None:
         """Unset the global pipeline instance when exiting a pipeline context."""
         _GlobalPipelineManager.set_pipeline(None)
+        if self.name == _PIPELINE_DEFAULT_NAME:
+            self.name = f"pipeline_{self._create_signature()[:8]}"
```

Comment:
what if instead of using `self._create_signature()`, we append the names of the step in the pipeline? With this solution, changing one parameter of one step will create another directory.

Reply:
Sure, makes more sense